youmubot-osu/src/discord/mod.rs

```diff
@@ -319,14 +321,46 @@ async fn add_user(
     user: models::User,
     data: &TypeMap,
 ) -> Result<()> {
+    let osu_client = data.get::<OsuClient>().unwrap();
+
+    let pp_fut = async {
+        [Mode::Std, Mode::Taiko, Mode::Catch, Mode::Mania]
+            .into_iter()
+            .map(|mode| async move {
+                osu_client
+                    .user(UserID::ID(user.id), |f| f.mode(mode))
+                    .await
+                    .map(|u_opt| u_opt.map(|u| u.pp.unwrap_or(0.0)))
+            })
+            .collect::<stream::FuturesOrdered<_>>()
+            .try_collect::<Vec<_>>()
+            .await
+            .unwrap()
```

Comment:
This would panic on API failure, pls handle gracefully 

(return `pp_fut` as Result and handle it or something)